src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php

```diff
@@ -113,4 +113,36 @@ private function executeSearchQuery($expectedResults = 1)
 
         return $results;
     }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRename()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin');
+
+        $result = $this->executeSearchQuery(1);
+        $entry = $result[0];
+        $this->assertEquals($entry->getAttribute('cn')[0], 'Kevin');
+    }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRenameWithoutRemovingOldRdn()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin', false);
+
+        $this->executeSearchQuery(2);
```

Comment:
Same thing. Don't forget to make sure that the new entry is removed after checking that you have two of them.

We do not want tests to put the directory in a state which differs from the startup.